app/backend/agents/ec2_agent.py

```diff
@@ -257,145 +300,274 @@ def run_instances(self, count, parameters, security_configured):
+    if spot == 'True':
         price = parameters[self.PARAM_SPOT_PRICE]
         conn.request_spot_instances(str(price), image_id, key_name=keyname,
           security_groups=[group], instance_type=instance_type, count=count, user_data = userstr)
-      else:
+    else:
         conn.run_instances(image_id, count, count, key_name=keyname,
           security_groups=[group], instance_type=instance_type, user_data=userstr)
-
-      instance_ids = []
-      public_ips = []
-      private_ips = []
-      utils.sleep(10)
-      end_time = datetime.datetime.now() + datetime.timedelta(0,
-        self.MAX_VM_CREATION_TIME)
-      now = datetime.datetime.now()
-
-      while now < end_time:
-        time_left = (end_time - now).seconds
-        utils.log('[{0}] {1} seconds left...'.format(now, time_left))
-        instance_info = self.describe_instances_old(parameters)
-        public_ips = instance_info[0]
-        private_ips = instance_info[1]
-        instance_ids = instance_info[2]
-        public_ips = utils.diff(public_ips, active_public_ips)
-        private_ips = utils.diff(private_ips, active_private_ips)
-        instance_ids = utils.diff(instance_ids, active_instances)
-        if count == len(public_ips):
-          break
-        time.sleep(self.SLEEP_TIME)
-        now = datetime.datetime.now()
-
-      if not public_ips:
-        self.handle_failure('No public IPs were able to be procured '
-                            'within the time limit')
-
-      if len(public_ips) != count:
-        for index in range(0, len(public_ips)):
-          if public_ips[index] == '0.0.0.0':
-            instance_to_term = instance_ids[index]
-            utils.log('Instance {0} failed to get a public IP address and' \
-                      ' is being terminated'.format(instance_to_term))
-            conn.terminate_instances([instance_to_term])
-
-      end_time = datetime.datetime.now()
-      total_time = end_time - start_time
-      if spot:
-        utils.log('TIMING: It took {0} seconds to spawn {1} spot ' \
-                  'instances'.format(total_time.seconds, count))
-      else:
-        utils.log('TIMING: It took {0} seconds to spawn {1} ' \
-                  'regular instances'.format(total_time.seconds, count))
-        if not skip_alarm:
-          utils.log('Creating Alarms for the instances')
-          for machineid in instance_ids:
-              self.make_sleepy(parameters, machineid)   
-      return instance_ids, public_ips, private_ips
-    except EC2ResponseError as exception:
-      self.handle_failure('EC2 response error while starting VMs: ' +
-                          exception.error_message)
-    except Exception as exception:
-      if isinstance(exception, AgentRuntimeException):
-        raise exception
-      else:
-        self.handle_failure('Error while starting VMs: ' + exception.message)
+        
+    return 
+    
+    
+    
+    
+    
+    
+    
+#     start_time = datetime.datetime.now()
```

Comment:
I will work on cleaning up the unnecessary comments and adding some necessary ones in the new merging part. 

Sent from my iPhone

> On Feb 4, 2015, at 13:48, Brian Drawert <notifications@github.com> wrote:
> 
> In app/backend/agents/ec2_agent.py:
> 
> > -                          exception.error_message)
> > -    except Exception as exception:
> > -      if isinstance(exception, AgentRuntimeException):
> > -        raise exception
> > -      else:
> > -        self.handle_failure('Error while starting VMs: ' + exception.message)
> > +        
> > +    return 
> > +    
> > +    
> > +    
> > +    
> > +    
> > +    
> > +    
> > +#     start_time = datetime.datetime.now()
> We should delete all this commented out code before it is merged.
> 
> —
> Reply to this email directly or view it on GitHub.
>